universe/Condition.cpp

```diff
@@ -8479,6 +8479,144 @@ unsigned int OrderedBombarded::GetCheckSum() const {
     return retval;
 }
 
+
+///////////////////////////////////////////////////////////
+// OrderedDestroyed                                      //
+///////////////////////////////////////////////////////////
+OrderedDestroyed::~OrderedDestroyed()
+{
+    delete m_by_object_condition;
+}
+
+bool OrderedDestroyed::operator==(const ConditionBase& rhs) const {
+    if (this == &rhs)
+        return true;
+    if (typeid(*this) != typeid(rhs))
+        return false;
+
+    const OrderedDestroyed& rhs_ = static_cast<const OrderedDestroyed&>(rhs);
+
+    CHECK_COND_VREF_MEMBER(m_by_object_condition)
+
+        return true;
+}
+
+namespace {
+    struct OrderedDestroyedSimpleMatch {
+        OrderedDestroyedSimpleMatch(const ObjectSet& by_objects) :
+            m_by_objects(by_objects)
+        {}
+
+        bool operator()(std::shared_ptr<const UniverseObject> candidate) const {
+            if (!candidate)
+                return false;
+            if (m_by_objects.empty())
+                return false;
+            std::shared_ptr<const Planet> planet = std::dynamic_pointer_cast<const Planet>(candidate);
+            if (!planet)
+                return false;
+            int planet_id = planet->ID();
+            if (planet_id == INVALID_OBJECT_ID)
+                return false;
+
+            // check if any of the by_objects is ordered to destroy the candidate planet
+            for (auto& obj : m_by_objects) {
+                auto ship = std::dynamic_pointer_cast<const Ship>(obj);
+                if (!ship)
+                    continue;
+                if (ship->OrderedDestroyPlanet() == planet_id)
+                    return true;
+            }
+            return false;
+        }
+
+        const ObjectSet& m_by_objects;
+    };
+}
+
+void OrderedDestroyed::Eval(const ScriptingContext& parent_context,
+    ObjectSet& matches, ObjectSet& non_matches,
+    SearchDomain search_domain/* = NON_MATCHES*/) const
+{
+    bool simple_eval_safe = parent_context.condition_root_candidate || RootCandidateInvariant();
+    if (simple_eval_safe) {
+        // evaluate contained objects once and check for all candidates
+        std::shared_ptr<const UniverseObject> no_object;
+        ScriptingContext local_context(parent_context, no_object);
+
+        // get subcondition matches
+        ObjectSet subcondition_matches;
+        m_by_object_condition->Eval(local_context, subcondition_matches);
+
+        EvalImpl(matches, non_matches, search_domain, OrderedDestroyedSimpleMatch(subcondition_matches));
+    }
+    else {
+        // re-evaluate contained objects for each candidate object
+        ConditionBase::Eval(parent_context, matches, non_matches, search_domain);
+    }
+}
+
+bool OrderedDestroyed::RootCandidateInvariant() const
+{
```

Comment:
move to end of prev line, or move actual code and closing `}` to same line as this